safe_autonomy_simulation/sims/inspection/sun.py

```diff
@@ -83,7 +83,7 @@ def state(self, state: np.ndarray):
         assert (
             state.shape == self.state.shape
         ), f"State shape must be {self.state.shape}, got {state.shape}"
-        self._state[0] = state[0]
+        self._state[-1] = state[0]
```

Comment:
This is why theta wasn't updating properly. The wrong index was being set to the new value.